### PR TITLE
Update docs on MetalLB LoadBalancer

### PR DESCRIPTION
**Pull request by NIXKnight**
MetalLB now uses a custom resource named `AddressPool` to define layer2 configuration. This PR updates the docs for it.

### CODE REVIEW

**Comment by aojea** [comment]
@fedepaol can you please review this?

**Comment by fedepaol** [comment]
@aojea @NIXKnight the current CRD in main (AddressPool) is an intermediate state to reach parity with the set of crds shipped by the metallb-operator. This means that with the next metallb release, AddressPool will be legacy and already obsoleted by IPPool (we couldn't maintain the name because of the 1:many relationship between the old and the new version).

What _I think_ makes sense at this stage is to pin the [metallb manifests](https://github.com/kubernetes-sigs/kind/blob/main/site/content/docs/user/loadbalancer.md?plain=1#L38) to the latest stable version (https://github.com/metallb/metallb/blob/v0.12.1/manifests/metallb.yaml).

**Comment by fedepaol** [comment]
Filed https://github.com/kubernetes-sigs/kind/pull/2685

File: site/static/examples/loadbalancer/metallb-addresspool.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: metallb.io/v1beta1
+kind: AddressPool
+metadata:
+  name: kind-addresspool
+  namespace: metallb-system
+spec:
+  protocol: layer2
+  addresses:
+    - 172.19.255.200-172.19.255.250
```
Comment (zaunist) [inline site/static/examples/loadbalancer/metallb-addresspool.yaml:9]:
```suggestion
    - 172.19.255.200-172.19.255.250

```
Would be better to add a black line?

Comment (NIXKnight) [reply site/static/examples/loadbalancer/metallb-addresspool.yaml:9]: sure.

**Comment by NIXKnight** [comment]
> @aojea @NIXKnight the current CRD in main (AddressPool) is an intermediate state to reach parity with the set of crds shipped by the metallb-operator. The second step is implemented in [this PR](https://github.com/metallb/metallb/pull/1245), plus there are a few things to bring in metallb, such as additional tests, webhooks, etc. This means that with the next metallb release, AddressPool will be legacy and obsoleted by IPPool (we couldn't maintain the name because of the 1:many relationship between the old and the new version).
> 
> What _I think_ makes sense at this stage is to pin the [metallb manifests](https://github.com/kubernetes-sigs/kind/blob/main/site/content/docs/user/loadbalancer.md?plain=1#L38) to the latest stable version (https://github.com/metallb/metallb/blob/v0.12.1/manifests/metallb.yaml).

@fedepaol got it. Do you suggest I close this PR then?

**Comment by fedepaol** [comment]
> > @aojea @NIXKnight the current CRD in main (AddressPool) is an intermediate state to reach parity with the set of crds shipped by the metallb-operator. The second step is implemented in [this PR](https://github.com/metallb/metallb/pull/1245), plus there are a few things to bring in metallb, such as additional tests, webhooks, etc. This means that with the next metallb release, AddressPool will be legacy and obsoleted by IPPool (we couldn't maintain the name because of the 1:many relationship between the old and the new version).
> > What _I think_ makes sense at this stage is to pin the [metallb manifests](https://github.com/kubernetes-sigs/kind/blob/main/site/content/docs/user/loadbalancer.md?plain=1#L38) to the latest stable version (https://github.com/metallb/metallb/blob/v0.12.1/manifests/metallb.yaml).
> 
> @fedepaol got it. Do you suggest I close this PR then?

Yep I think it makes sense for the time being. Will try to remember to update here once things settle in metallb and we do a formal release.

**Comment by NIXKnight** [comment]
@fedepaol ok I am closing this pull request. Thanks for the info.

**Pull request closed without merging**